modules/document_repository/ajax/documentEditUpload.php

```diff
@@ -120,7 +115,7 @@
         $comments   = $_POST['commentsEdit'];
         $version    = $_POST['versionEdit'];
 
-        if (empty($category) && $category !== '0') {
```

Comment:
`empty` checks for `"0"` so this is redundant.

Reply:
This seems to be purposefully checking for a 0 string. You are changing behavior here. 

Ie if categories = 0, ur change makes no difference. But is categories ='0', the original code will evaluate to false while urs will evaluate to true. It is not redundant in this case

Reply:
Oh I see. I read that wrong. Good catch

Reply:
I've changed this check to cast `$category` as an int and check whether its value is >= 0, and added a note to this effect. This makes it more clear what's happening, imo.

Reply:
I feel like a comment could have made it more clear, personally not a fan of the int cast, but also i don't care enough to request a change 😃

Reply:
@ZainVirani I decided to use `is_numeric` so that this would be more clear

Reply:
Added a comment so we get the best of all possible worlds